src/hooks/useGetUserPatchesPageTitleAndLink.ts

```diff
@@ -0,0 +1,34 @@
+import { GET_OTHER_USER } from "gql/queries";
+import {
+  GetOtherUserQuery,
+  GetOtherUserQueryVariables,
+} from "gql/generated/types";
+import { useQuery } from "@apollo/react-hooks";
+import { getUserPatchesRoute } from "constants/routes";
+
+export const useGetUserPatchesPageTitleAndLink = (userId: string) => {
+  const { data, loading, error } = useQuery<
+    GetOtherUserQuery,
+    GetOtherUserQueryVariables
+  >(GET_OTHER_USER, { variables: { userId } });
+  const link = getUserPatchesRoute(userId);
+
+  if (loading || error) {
+    return { error, link, loading, title: "Patches" };
+  }
+
+  if (userId === data?.currentUser.userId) {
+    return { error, link, loading, title: "My Patches" };
+  }
+
+  const otherUserDisplayName = data?.otherUser.displayName ?? "";
```

Comment:
can we do `data?.otherUser?.displayName` to be extra careful on avoiding type errors in prod?

Reply:
this is already type safe since the result of the `user` gql query is non-nullable